include/alpaka/core/BoostPredef.hpp

```diff
@@ -12,26 +13,24 @@
 #endif
 
 //---------------------------------------HIP-----------------------------------
-// __HIPCC__ is defined by hipcc (if either __CUDACC__ is defined)
+// __HIPCC__ is defined by hipcc when either __CUDACC__, __HIP__ or __HCC__ are defined.
 // https://github.com/ROCm-Developer-Tools/HIP/blob/master/docs/markdown/hip_porting_guide.md#compiler-defines-summary
 #if !defined(BOOST_LANG_HIP)
-#    if defined(__HIPCC__) && (defined(__CUDACC__) || defined(__HIP__))
-#        include <hip/hip_runtime.h>
-// HIP defines "abort()" as "{asm("trap;");}", which breaks some kernels
-#        undef abort
+#    if defined(__HIPCC__)
+/* BOOST_LANG_CUDA is enabled when either __CUDACC__ (nvcc) or __CUDA__ (clang) are defined. This occurs when
+   nvcc / clang encounter a CUDA source file. Since there are no HIP source files we treat every source file
+   as HIP when we are using a HIP-capable compiler. */
+#        include <hip/hip_version.h>
+// HIP doesn't give us a patch level for the last entry, just a gitdate
 #        define BOOST_LANG_HIP BOOST_VERSION_NUMBER(HIP_VERSION_MAJOR, HIP_VERSION_MINOR, 0)
-#        if defined(BOOST_LANG_CUDA) && BOOST_LANG_CUDA
-#            undef BOOST_LANG_CUDA
-#            define BOOST_LANG_CUDA BOOST_VERSION_NUMBER_NOT_AVAILABLE
-#        endif
 #    else
 #        define BOOST_LANG_HIP BOOST_VERSION_NUMBER_NOT_AVAILABLE
 #    endif
 #endif
 
 // HSA device architecture detection (HSA generated via HIP(clang))
 #if !defined(BOOST_ARCH_HSA)
-#    if defined(__HIP_DEVICE_COMPILE__) && __HIP_DEVICE_COMPILE__ == 1 && defined(__HIP__)
+#    if defined(__HIP_DEVICE_COMPILE__) && __HIP_DEVICE_COMPILE__ == 1
```

Comment:
`__HIP_DEVICE_COMPILE__` implies `__HIP__` since we don't support other HIP compilers anymore.

Reply:
As I wrote above please keep `__HIP__` to not remove vanilla clang support.